### PR TITLE
Remove callback

### LINKED ISSUE #1: Promisify
Is this something you would like to have promisified? Or is this a ponyfill?

### PR DESCRIPTION
**Pull request by SamVerschueren**
Resolves #1 

Added XO as well.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,19 +1,4 @@
...
-			cb(err);
-		});
-
-		return false;
-	}
-
-	return true;
-}
+module.exports = function (pth) {
+	return String(pth).indexOf('\u0000') > -1;
```
Comment (sindresorhus) [inline index.js]: Wouldn't it make more sense if it still threw? The whole point of this module is not having to redefine the error message and code.

Comment (SamVerschueren) [inline index.js]: So, throw error if null check fails, else return true?

Comment (sindresorhus) [inline index.js]: It doesn't need to return anything.

**Comment by SamVerschueren** [comment]
Hopefully this is what you mean :).

File: readme.md [hunk 2]
```diff
@@ -15,17 +13,10 @@ $ npm install --save null-check
...
-	//=> { [Error: Path must be a string without null bytes.] code: 'ENOENT' }
-});
-//=> false
+const nullCheck = require('null-check');
 
-// the method is sync without a callback
-nullCheck('unicorn.png');
-//=> true
+nullCheck('unicorn.png\u0000');
+//=> throws `Path must be a string without null bytes.`
```
Comment (SamVerschueren) [inline readme.md]: Should we surround with try-catch instead?

``` js
try {
    nullCheck('unicorn.png\u0000');
} catch (err) {
    // => `Path must be a string without null bytes.`
}
```

Comment (sindresorhus) [inline readme.md]: Sure

**Comment by SamVerschueren** [comment]
Let me know if you're ok with the changes, then I'll squash the commits.

File: test.js [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
-test('async', function (t) {
-	t.plan(2);
-
-	nullCheck('unicorn.png\u0000', function (err) {
-		t.assert(err, err);
-		console.log(err)
-		t.assert(err.code === 'ENOENT');
-	});
+test('null bytes', t => {
+	try {
```
Comment (sindresorhus) [inline test.js]: use `t.throws`

File: test.js [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+		t.is(err.code, 'ENOENT');
+	} finally {
+		t.end();
+	}
 });
 
-test('sync', function (t) {
-	t.assert(nullCheck('unicorn.png'));
+test('no null bytes', t => {
+	fn('unicorn.png');
```
Comment (sindresorhus) [inline test.js]: use `t.doesNotThrow`

File: readme.md [hunk 5]
```diff
@@ -15,17 +13,13 @@ $ npm install --save null-check
...
-//=> false
+const nullCheck = require('null-check');
 
-// the method is sync without a callback
-nullCheck('unicorn.png');
-//=> true
+try {
+	nullCheck('unicorn.png\u0000');
+} catch (err) {
+	//=> `Path must be a string without null bytes.`
```
Comment (sindresorhus) [inline readme.md]:
```
console.log(err.message);
//=> 'Path must be a string without null bytes.'
```

Note the single-quotes

File: test.js (hunk 3 above)
Comment (SamVerschueren) [inline test.js]: Did to much async/await stuff I guess :p

Comment (sindresorhus) [inline test.js]: I figured :p

**Comment by SamVerschueren** [comment]
This should be it I guess? Squash them all up into one commit?

**Comment by SamVerschueren** [comment]
Awesome, thanks for merging! :beers:

**Comment by sindresorhus** [comment]
:tropical_drink:

**Pull request closed without merging**